Mathlib/Geometry/Euclidean/Similarity.lean

```diff
@@ -0,0 +1,157 @@
+    ![a, b, c] ∼ ![a', b', c'] := by
+  have dist_a'b' : dist a' b' ≠ 0 := by simp [ne₁₂_of_not_collinear h_not_col']
+  have dist_b'c' : dist b' c' ≠ 0 := by simp [ne₂₃_of_not_collinear h_not_col']
+  rw [← div_eq_div_iff dist_a'b' dist_b'c'] at hd
+  set k := (dist a b / dist a' b') with hk
+  have k_ne : 0 < k := by
+    rw [hk]
+    apply div_pos
+    · simp [dist_pos, ne₁₂_of_not_collinear h_not_col]
+    · simp [dist_pos, ne₁₂_of_not_collinear h_not_col']
+  have h_ab : dist a b = k * dist a' b' := by
+    rw [hk, div_mul, div_self dist_a'b', div_one]
+  have h_bc : dist b c = k * dist b' c' := by
+    rw [hd, div_mul, div_self dist_b'c', div_one]
+  have hcos := law_cos a b c
+  rw [dist_comm b _ , dist_comm b' _] at h_bc
+  rw [h_ab, h_bc] at hcos
+  field_simp at hcos
+  rw [h] at hcos
+  have hcos' := law_cos a' b' c'
+  field_simp at hcos'
+  rw [← hcos', ← mul_pow] at hcos
+  have dist_ac : 0 < dist a c := by
+    simp [dist_pos, ne₁₃_of_not_collinear h_not_col]
+  have dist_a'c' : 0 < dist a' c' := by
+    simp [dist_pos, ne₁₃_of_not_collinear h_not_col']
+  have k_dist_a'c' : 0 ≤ k * dist a' c' := by positivity
+  rw [pow_left_inj₀ (le_of_lt dist_ac) k_dist_a'c' (by norm_num)] at hcos
+  rw [dist_comm c _, dist_comm c' _] at h_bc
+  rw [dist_comm a _, dist_comm a' _] at hcos
+  rw [similar_iff_exists_pos_pairwise_dist_eq]
+  use k
+  refine ⟨k_ne, ?_⟩
+  intro i j hij
+  fin_cases i <;> fin_cases j <;> try {rw [dist_self, dist_self, mul_zero]}
+  all_goals simp; grind [dist_comm]
+
+/-- For two similar triangles, the corresponding angles are equal. -/
+theorem angle_eq_of_similar (h : ![a, b, c] ∼ ![a', b', c']) :
+    ∠ a b c = ∠ a' b' c' := by
+  rw [similar_iff_exists_pos_dist_eq] at h
+  rcases h with ⟨r, hr_pos, hdist⟩
+  have h_ab : dist a b = r * dist a' b' := hdist 0 1
+  have h_cb : dist c b = r * dist c' b' := hdist 2 1
+  have h_ac : dist a c = r * dist a' c' := hdist 0 2
+  have h_cos := law_cos a b c
+  rw [h_ab, h_cb, h_ac] at h_cos
+  field_simp at h_cos
+  have h_cos' := law_cos a' b' c'
+  field_simp at h_cos'
+  rw [h_cos', sub_right_inj] at h_cos
+  by_cases heq : dist a' b' * dist c' b' * 2 = 0
+  · rw [mul_eq_zero_iff_right (by norm_num), mul_eq_zero] at heq
+    rcases heq with h1 | h2
+    · have h_dist_ab : dist a b = 0 := by
+        rw [h_ab, h1, mul_zero]
+      rw [dist_eq_zero] at h_dist_ab h1
+      simp_rw [h_dist_ab, h1, angle_self_left]
+    · have h_dist_cb : dist c b = 0 := by
+        rw [h_cb, h2, mul_zero]
+      rw [dist_eq_zero] at h_dist_cb h2
+      simp_rw [h_dist_cb, h2, angle_self_right]
+  rw [mul_right_inj' heq] at h_cos
+  apply Real.injOn_cos at h_cos
+  · rw [h_cos]
+  repeat exact ⟨angle_nonneg _ _ _, angle_le_pi _ _ _ ⟩
+
+/-- In two similar triangles, all three corresponding angles are equal. -/
+theorem Similar.angle_eq_all (h : ![a, b, c] ∼ ![a', b', c']) :
+    ∠ a b c = ∠ a' b' c' ∧ ∠ b c a = ∠ b' c' a' ∧ ∠ c a b = ∠ c' a' b' := by
```

Comment:
And as you're inside the `EuclideanGeometry` namespace, this needs to be given as `_root_.Similar.angle_eq_all` to work as intended.